### DISCUSSION TITLE
Additional Macros in Mainsail or Fluid

### DISCUSSION THREAD

**Discussion by Avpman2** [xblax/flashforge_ad5m_klipper_mod#176] [Q&A] [answered]
Sorry, "almost" not a NOOB any longer. But is there a way for me to add/remove additional macros to the Mainsail or Fluid interface?

**Comment by Avpman2** [comment]
I located how to disable certain macros but not how to add new ones. Anyone?

**Reply by consp** [reply]
> not how to add new ones

Edit macros.cfg in mainsail or fluidd (located in /root/printer_data/configs). You can do that via the web gui.

![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/6802998/9bb86d36-2323-461c-9aa4-c41829af9c11)

edit the file to your liking:

![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/6802998/28dd4076-4227-4b55-a565-318442860cf0)

**Comment by Avpman2** [comment]
Thanks!

I can’t find in the GUI where to edit the cfg file. I found how to show/hide macros but not where to edit the cfg file.

**Reply by consp** [reply, marked as answer]
See the previous image (click where the red "circles" point to) in fluidd or if you are using mainsail:

![image](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/6802998/15d1e8ee-b1c2-4676-8828-bdfa2f7ed4c1)

**Comment by Avpman2** [comment]
Awesome, thank you!

**Comment by Avpman2** [comment]
I did as indicated and checked twice. My new macro is in the macro.cfg file, but I still can't select it to appear in the Devices Tab of Orcaslicer. Basically, I copied the code under [gcode_macro START_PRINT] code to [gcode_macro START_PRINT_2]

**Reply by consp** [reply]
Did you restart klipper after you updated it (enter RESTART in the console in mainsail/fluidd)? You can of course test the macro by by running it from the [console](https://docs.mainsail.xyz/overview/features/console) without having to use the slicer.
